### PR TITLE
Restore commented-out load tests

### PR DESCRIPTION
**Pull request by bastbu**
# PR for issue #1160

## What is being addressed

With this PR we restore (or remove) the commented-out load tests and ensure that they can run successfully in the CI.

## How is this addressed

We uncomment and refactor the existing load tests.
- The simulation test which tests the decoder we remove, since it does not add value over our E2E test and will only make the simulator setup harder when running locally
- The actual load test we uncomment and refactor - we homogenize the test with the helper methods of the other load tests that we have, and we remove assertions that we now cover through other assertions (e.g. we do not test the exact message any longer, but just that the IoT hub received N messages from a given device)
- We stabilize the assertions by allowing some receive windows to be missed based on a configurable parameter

### CODE REVIEW

File: Tests/Simulation/IntegrationTestFixtureSim.cs
Comment (spygi) [inline Tests/Simulation/IntegrationTestFixtureSim.cs:155]: Not sure if it's an issue but with the new helper method `CreateOtaaDevice` no GatewayId is assigned.

File: Tests/Simulation/IntegrationTestFixtureSim.cs
Comment (spygi) [inline Tests/Simulation/IntegrationTestFixtureSim.cs:96]: Did you consider reusing the methods `CreateABPDevice` and `CreateOTAADevice` from `TestDeviceInfo` here?

File: Tests/Simulation/SimulatedLoadTests.cs
Comment (spygi) [inline Tests/Simulation/SimulatedLoadTests.cs:176]: [nit] Is there a technical reason why you try to send messages for both ABP and OTAA devices in a single for-loop? I find that it complicates the parameters we need to validate (e.g. messagesBeforeConfirmed <= messagesBeforeJoin) and the logic (e.g. for OTAA if messageId < messagesBeforeJoin + initialMessageId we don't send anything)

File: Tests/Simulation/SimulatedLoadTests.cs
Comment (spygi) [inline Tests/Simulation/SimulatedLoadTests.cs:194]: Not sure about this but method declarations should be cached, right? Asking because this is inside a for-loop.

**Review by spygi — COMMENTED** [review]
LGTM - left some suggestions, waiting for approval as per our discussion.

File: Tests/Simulation/IntegrationTestFixtureSim.cs
Comment (bastbu) [reply Tests/Simulation/IntegrationTestFixtureSim.cs:155]: Eagle eyes :) It was intentional though, since for the multi-gateway scenarios we do not want to associate these devices with a gateway already.

File: Tests/Simulation/SimulatedLoadTests.cs
Comment (bastbu) [reply Tests/Simulation/SimulatedLoadTests.cs:176]: In any case I would need to duplicate a loop (if I send ABP and OTAA separately it would need to be the loop over message IDs). Now, to make sure that I send OTAA and ABP for the same message ID at the same time, it needs to be in the same loop. We could argue whether it would be good enough to send OTAA and ABP after each other, but I decided to just draw a line somewhere and leave the old test as it was (with respect to its "flow").

No strong opinion here though, we can discuss whether the message ID is still a useful concept for that test at all.

File: Tests/Simulation/SimulatedLoadTests.cs
Comment (bastbu) [reply Tests/Simulation/SimulatedLoadTests.cs:194]: Checked this in Sharplab, it is indeed converted to just a class method. I still moved the static local functions out of the loop, it is less distracting indeed.

File: Tests/Simulation/SimulatedLoadTests.cs
Comment (bastbu) [reply Tests/Simulation/SimulatedLoadTests.cs]: Valid point, done this in 6c71306eb99608263b5c4a034d2afde5698440a5

File: Tests/Simulation/IntegrationTestFixtureSim.cs
Comment (bastbu) [reply Tests/Simulation/IntegrationTestFixtureSim.cs:96]: I did, but to be honest I was not confident what I would break by reusing the mentioned methods. There are some specifics in there around net ids etc that I did not want to change as part of this PR, I did the least effort fix, which was extracting the duplicated code to a local function. Open to improve this if you have more insights about what the changes would imply.

File: Tests/Simulation/SimulatedLoadTests.cs
Comment (bastbu) [reply Tests/Simulation/SimulatedLoadTests.cs]: As per our discussion, I get it now. Originally, I had it as you suggested, but that did not work as expected due to some details around materialization of the Enumerables

Comment (spygi) [reply Tests/Simulation/SimulatedLoadTests.cs]: I see, in that case I think we can resolve, it was more of a nit comment in any case.

**Review by spygi — APPROVED** [review]

**Pull request merged**
